internal/devbox/devopt/devboxopts.go

```diff
@@ -76,4 +75,10 @@ type EnvOptions struct {
 	OmitNixEnv        bool
 	PreservePathStack bool
 	Pure              bool
+	RecomputeEnv      *RecomputeEnvOpts
```

Comment:
Why pointer? Default to structs unless you a) need mutating methods b) struct is massive and you want to avoid copying data.

Reply:
In general, I go with pointers because its easier for the zero value when writing instantiations of the struct. This is especially relevant for test-cases. The downside is that the caller has to be more defensive for the nil-case.

Reply:
This file uses all non-pointers, so in the interest of consistency we should keep that.

I don't follow why zero value is easier with non-pointer. Structs default to their zero values, no need to initialize anything.

The downside is quite large. A nil pointer use is a panic.